Tests/Swift Tests/BugsnagSwiftPublicAPITests.swift

```diff
@@ -0,0 +1,368 @@
+        client.context = ""
+        _ = client.context
+        
+        let _ = client.appDidCrashLastLaunch()
+        
+        client.setUser("me", withEmail: "memail@foo.com", andName: "you")
+        let _ = client.user()
+        
+        client.addOnSession(block: sessionBlock)
+        client.removeOnSession(block: sessionBlock)
+        
+        client.addOnSendError(block: onSendErrorBlock)
+        client.removeOnSendError(block: onSendErrorBlock)
+        
+        client.addOnBreadcrumb(block: onBreadcrumbBlock)
+        client.removeOnBreadcrumb(block: onBreadcrumbBlock)
+    }
+
+    func testBugsnagDeviceClass() throws {
+        let device = BugsnagDevice()
+        
+        device.jailbroken = false
+        _ = device.jailbroken
+        
+        device.id = nil
+        device.id = "id"
+        _ = device.id
+        
+        device.locale = nil
+        device.locale = "locale"
+        _ = device.locale
+        
+        device.manufacturer = nil
+        device.manufacturer = "man"
+        _ = device.manufacturer
+        device.model = nil
+        device.model = "model"
+        _ = device.model
+        device.modelNumber = nil
+        device.modelNumber = "model"
+        _ = device.modelNumber
+        device.osName = nil
+        device.osName = "name"
+        _ = device.osName
+        device.osVersion = nil
+        device.osVersion = "version"
+        _ = device.osVersion
+        device.runtimeVersions = nil
+        device.runtimeVersions = [:]
+        device.runtimeVersions = ["a" : "b"]
+        _ = device.runtimeVersions
+        device.totalMemory = nil
+        device.totalMemory = 1234
+        _ = device.totalMemory
+    }
+
+    func testBugsnagEndpointConfigurationlass() throws {
+        let epc = BugsnagEndpointConfiguration()
+        epc.notify = "notify"
+        epc.sessions = "sessions"
+    }
+
+    // Also error types
+    func testBugsnagErrorClass() throws {
+        let e = BugsnagError()
+        e.errorClass = nil
+        e.errorClass = "class"
+        _ = e.errorClass
+        e.errorMessage = nil
+        e.errorMessage = "msg"
+        _ = e.errorMessage
+        
+        e.type = .cocoa
+        e.type = .c
+        e.type = .reactNativeJs
+    }
+
+    func testBugsnagSessionClass() throws {
+        let session = BugsnagSession()
+        session.id = "id"
+        _ = session.id
+        session.startedAt = Date()
+        _ = session.startedAt
+        _ = session.app
+        _ = session.device
+        _ = session.user
+        session.setUser("user", withEmail: "email", andName: "name")
+    }
+
+    func testBugsnagStackframeClass() throws {
+        let sf = BugsnagStackframe()
+        sf.method = nil
+        sf.method = "method"
+        _ = sf.method
+        sf.machoFile = nil
+        sf.machoFile = "file"
+        _ = sf.machoFile
+        sf.machoUuid = nil
+        sf.machoUuid = "uuid"
+        _ = sf.machoUuid
+        sf.frameAddress = 123
```

Comment:
This can now be nil as of #614, along with additional properties in that changeset